habitat-lab/habitat/gui/gui_application.py

```diff
@@ -150,7 +150,7 @@ def _post_sim_update(self, post_sim_update_dict):
         if "application_cursor" in post_sim_update_dict:
             self.cursor = post_sim_update_dict["application_cursor"]
 
-        if "application_exit" in post_sim_update_dict:
+        if post_sim_update_dict.get("should_exit", False):
```

Comment:
I prefer `application_exit`. For messages that specifically go to `GuiApplication`, I was trying to make that clear by using the `application` prefix in the name. If you feel strongly, you could rename to `application_should_exit`.

Reply:
Also how is your version more Pythonic or cleaner (doing a `get` versus `in`)? It seems less Pythonic.

Reply:
> I prefer application_exit. For messages that specifically go to GuiApplication

👍 , I'll revert this key change, makes sense

> Also how is your version more Pythonic or cleaner (doing a get versus in)? It seems less Pythonic.

Earlier `application_exit` was added to post_sim_update_dict only if application exit was initiated (ESC pressed). After that,  exit app condition was evaluated by checking if `"application_exit"` key is present in `post_sim_update_dict`. This was, `application_exit` key's value was never used. In this implementation I check `application_exit` key's value that's why I used `get` method. It seems to me that it is better to have `post_sim_update_dict` with fixed keys instead of variable. For example, if we ever use `application_exit` key then app behaviour should depend on its value not key being present in the dict or not. But I don't feel strongly about this, can undo this change if you wish.